### PR TITLE
Add coriolis-assess-migration and instance utils.

### PR DESCRIPTION
**Pull request by atoaca**

### CODE REVIEW

File: coriolis_openstack_utils/assess_instance_cmd.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        # oslo_conf error out with urecognized arguments:
+        ["--config-file", args.conf_file],
+        project=constants.PROJECT_NAME,
+        version=constants.PROJECT_VERSION)
+
+    source_client = conf.get_source_openstack_client()
+    instance_names = args.instances
+    result = instances.instances_assess(source_client, instance_names)
+    if args.format:
+        if args.format.lower() == "yaml":
```
Comment (aznashwan) [inline coriolis_openstack_utils/assess_instance_cmd.py]: you can declare allowed values for this argument in the parser

File: coriolis_openstack_utils/assess_migration_cmd.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        project=constants.PROJECT_NAME,
+        version=constants.PROJECT_VERSION)
+
+    migration_id = args.migration_id
+    source_client = conf.get_source_openstack_client()
+    coriolis = conf.get_coriolis_client()
+    result = instances.migration_assess(source_client, coriolis, migration_id)
+
+    if args.format:
+        if args.format.lower() == "yaml":
```
Comment (aznashwan) [inline coriolis_openstack_utils/assess_migration_cmd.py]: ditto

File: coriolis_openstack_utils/instances.py [hunk 3]
```diff
@@ -120,3 +123,99 @@ def validate_migration_options(
         LOG.info("%s volume types are not mapped." %
                  instance_volume_types -
                  source_mapped_volume_types)
+
+
+def instances_assess(source_client, instances_names):
```
Comment (aznashwan) [inline coriolis_openstack_utils/instances.py]: Vague/unrelated function name.
`def get_instances_assessment`

Comment (aznashwan) [reply coriolis_openstack_utils/instances.py]: Also, split this into a function which only takes one instance name as an argument; and then just loop through all the names and call it here.

File: coriolis_openstack_utils/instances.py [hunk 4]
```diff
@@ -120,3 +123,99 @@ def validate_migration_options(
...
+
+    interval_date = finish_date - creation_date
+
+    migration_instances = migration.instances
+    assessment_list = instances_assess(source_client, migration_instances)
+    for assessment in assessment_list:
+        assessment["migration"] = {}
+        assessment["migration"]["migration_id"] = migration.id
+        assessment["migration"]["migration_status"] = migration.status
+        assessment["migration"]["migration_time"] = str(interval_date)
```
Comment (aznashwan) [inline coriolis_openstack_utils/instances.py:237]: how does the time diff look when cast to a string?

File: coriolis_openstack_utils/schemas/instance_assess_info.json [hunk 5]
```diff
@@ -0,0 +1,106 @@
+{
+  "$schema": "http://cloudbase.it/coriolis-utils/schemas/instance_assessment#",
```
Comment (aznashwan) [inline coriolis_openstack_utils/schemas/instance_assess_info.json:2]: Please add `additionalProperties: false` to all items of type 'object' (google it)

File: coriolis_openstack_utils/instances.py (hunk 4 above)
Comment (atoaca) [reply coriolis_openstack_utils/instances.py:237]: H:M:S, thankfully.

File: coriolis_openstack_utils/instances.py [hunk 6]
```diff
@@ -120,3 +124,111 @@ def validate_migration_options(
...
+    cinder = source_client.cinder
+
+    assessment = {}
+    assessment['storage'] = {}
+    total_size_gb = 0
+    if instance.image:
+        # Taking in account that the source image might be deleted
+        try:
+            image = glance.images.get(instance.image.get('id'))
+            image_size = math.ceil(image.size / (1024 ** 3))
```
Comment (aznashwan) [inline coriolis_openstack_utils/instances.py]: please use `oslo_utils.units.Gi`

File: coriolis_openstack_utils/instances.py [hunk 7]
```diff
@@ -120,3 +124,111 @@ def validate_migration_options(
...
+        except ImageNotFound:
+            pass
+
+    volume_ids = [vol.id for vol in
+                  nova.volumes.get_server_volumes(instance.id)]
+    volumes = [cinder.volumes.find(id=vol_id) for vol_id in volume_ids]
+    total_size_gb += sum([volume.size for volume in volumes])
+    volumes_info = [{"volume_name": volume.name,
+                     "volume_id": volume.id,
+                     "size_bytes": 2**30 * volume.size}
```
Comment (aznashwan) [inline coriolis_openstack_utils/instances.py]: please use `oslo_utils.units.Gi`

File: coriolis_openstack_utils/instances.py [hunk 8]
```diff
@@ -120,3 +124,111 @@ def validate_migration_options(
...
+                    for volume in volumes]
+    assessment["storage"]["volumes"] = volumes_info
+
+    instance_flavor = nova.flavors.get(instance.flavor['id'])
+    total_size_gb += instance_flavor.disk
+    flavor_info = {"flavor_name": instance_flavor.name,
+                   "flavor_id": instance_flavor.id,
+                   "flavor_vcpus": instance_flavor.vcpus,
+                   "flavor_disk_size":
+                   instance_flavor.disk * (2 ** 30)}
```
Comment (aznashwan) [inline coriolis_openstack_utils/instances.py]: oslo_utils.units

**Review by aznashwan — COMMENTED** [review]
Always consider using `oslo_utils` in any OpenStack-related projects.
Among its submodules, you can find common units/multiples/submultiples in `units` (ex: `oslo_utils.units.Gi`)
It may seem overkill to import a whole module for 3 constants, but those are the exact same constants ALL the OpenStack projects are using, so we can be 100% certain of compatibility.

Please update the PR to make use of those constants.
You might need to add `oslo_utils` in `requirements.txt`

**Pull request closed without merging**